src/structures/GuildAuditLogs.js

```diff
@@ -383,16 +383,19 @@ class GuildAuditLogsEntry {
       case Actions.CHANNEL_OVERWRITE_CREATE:
       case Actions.CHANNEL_OVERWRITE_UPDATE:
       case Actions.CHANNEL_OVERWRITE_DELETE:
-        switch (data.options.type) {
-          case 'member':
-            this.extra = guild.members.cache.get(data.options.id) || { id: data.options.id, type: 'member' };
-            break;
-
-          case 'role':
+        switch (OverwriteTypes[data.options.type]) {
+          case OverwriteTypes[0]:
```

Comment:
Ok, has this been tested? Does it work as intended? Because `OverwriteTypes[0] === 'role'` and `OverwriteTypes.role === 0`...

Honestly this whole check is confusing as heck. So we're switching by the.. string representation of the type and setting the type to the number???

that seems awfully inefficient if so. A simple `switch (data.options.type)` with `case OverwriteTypes.role` and such MIGHT be better overall, but also setting the `this.extra.type` to the string variant

Reply:
iirc i did it like that because when fetching audit logs, you get a string as the type instead of the number, but the string is the number, if that makes any sense, discord sends `'0'` instead of `0` here